packages/internal-workflow-common/src/activity-options.ts

```diff
@@ -87,6 +80,80 @@ Note that the Temporal Server doesn't detect Worker process failures directly. I
   cancellationType?: coresdk.workflow_commands.ActivityCancellationType;
 }
 
+/**
+ * Options for local activity invocation
+ */
+export interface LocalActivityOptions {
+  /**
+   * Identifier to use for tracking the activity in Workflow history.
+   * The `activityId` can be accessed by the activity function.
+   * Does not need to be unique.
+   *
+   * @default an incremental sequence number
+   */
+  activityId?: string;
+
+  /**
+   * RetryPolicy that define how activity is retried in case of failure. If this is not set, then the server-defined default activity retry policy will be used. To ensure zero retries, set maximum attempts to 1.
+   */
+  retry?: RetryPolicy;
+
+  /**
+   * Maximum time the local activity is allowed to execute after the task is dispatched. This
+   * timeout is always retryable.
+   *
+   * Either this option or {@link scheduleToCloseTimeout} is required.
+   * If set, this must be <= {@link scheduleToCloseTimeout}, otherwise, it will be clamped down.
+   *
+   * @format {@link https://www.npmjs.com/package/ms | ms} formatted string or number of milliseconds
+   */
+  startToCloseTimeout?: string | number;
+
+  /**
+   * Limits time the local activity can idle internally before being executed. That can happen if
+   * the worker is currently at max concurrent local activity executions. This timeout is always
+   * non retryable as all a retry would achieve is to put it back into the same queue. Defaults
+   * to {@link scheduleToCloseTimeout} if not specified and that is set. Must be <=
+   * {@link scheduleToCloseTimeout} when set, otherwise, it will be clamped down.
+   *
+   * @default unlimited
+   * @format {@link https://www.npmjs.com/package/ms | ms} formatted string or number of milliseconds
+   */
+  scheduleToStartTimeout?: string | number;
```

Comment:
This doesn't exist for local activities in Java and Go (but if it has value in TS, cool)

Reply:
It's of dubious value, but, I implemented it mostly because it lets us more easily combine LA/nonLA eventually. IMO if we just don't expose it in TS for now that's completely legit.

Reply:
Let's keep this for uniformity.